### ISSUE TITLE
commands fail unhelpfully when `cliSetupTritonApi` returns error

### ISSUE THREAD

**Issue by gwydirsam** [joyent/node-triton#213]
When `cliSetupTritonApi` calls its callback with an error, the code reads:

```js
    common.cliSetupTritonApi({cli: this}, function onSetup(setupErr) {
        if (setupErr) {
            callback(setupErr);
        }
        /* .. continues */
```

There should be a `return` in the block under `if (setupErr)` -- otherwise the code proceeds to try to use the `tritonapi.cloudapi` and throws out a stack trace immediately, and then never gets to print the actual error that came out of `cliSetupTritonApi`.

This can be seen with e.g. `triton ls`, with an ssh agent error:

# To reproduce:

```bash
bash-3.2$ ssh-add -D; SDC_KEY_ID="ab:cd:ef" triton ls
/opt/pkg/lib/node_modules/triton/lib/do_instance/do_list.js:109
                self.top.tritonapi.cloudapi.listMachines(
                                           ^

TypeError: Cannot read property 'listMachines' of undefined
    at Array.getTheMachines (/opt/pkg/lib/node_modules/triton/lib/do_instance/do_list.js:109:44)
    at Object.parallel (/opt/pkg/lib/node_modules/triton/node_modules/vasync/lib/vasync.js:107:11)
    at onSetup (/opt/pkg/lib/node_modules/triton/lib/do_instance/do_list.js:81:16)
    at /opt/pkg/lib/node_modules/triton/lib/common.js:776:13
    at /opt/pkg/lib/node_modules/triton/lib/tritonapi.js:375:17
    at /opt/pkg/lib/node_modules/triton/node_modules/smartdc-auth/lib/keyring.js:168:4
    at /opt/pkg/lib/node_modules/triton/node_modules/smartdc-auth/lib/keyring.js:142:4
    at /opt/pkg/lib/node_modules/triton/node_modules/smartdc-auth/node_modules/vasync/lib/vasync.js:81:5
    at /opt/pkg/lib/node_modules/triton/node_modules/smartdc-auth/lib/keyring.js:153:5
    at /opt/pkg/lib/node_modules/triton/node_modules/smartdc-auth/lib/kr-homedir.js:102:4
```

**Comment by gwydirsam** [comment]
Turns out smartdc-auth does not support finding an ecdsa key by it's fingerprint.

**Comment by arekinath** [comment]
https://github.com/joyent/node-triton/issues/213

**Comment by trentm** [comment]
This is published in triton@5.2.1
